### ISSUE TITLE
Purpose

### ISSUE THREAD

**Issue by ne9z** [ne9z/dotfiles-flake#7]
Hello, and thanks for taking interest in my work.  A brief history of
my involvement with Root on ZFS is the following.

I started to host a small file server for my family in 2018.  With too
much free time on hand, I thought it would be a good idea to provide
redundancy in the setup.  This includes two aspects:

- Protection against data loss caused by disk failure, and
- Continuity of operation; that is, the system should still be
  bootable even if one disk fails.

As many other people before me have done, I came to the conclusion
that ZFS would fit the bill.  To make the second goal possible, Root
on ZFS is necessary.

At that time, only Debian/Ubuntu guides were available.  I tried to
follow them, but the process was quite time-consuming and error-prone,
due in no small part to my total lack of experience of Linux
command line.  Later, as I distro-hopped, I wrote guides for Alpine,
Arch, and NixOS.

Looking back, I realized the purpose of the installation guides.  They
are like normal computer programs, or a piece of hardware.

As long as the instructions are clear and complete, the user should
not be concerned about how or why the guide work, just like you are
not too concerned about how or why

- the internet browser,
- computer screen, or
- an airplane

work, as long as they are able to

- play a video,
- display a picture, or
- take you to Europe.

Once the user completed their system installation, they should move on
to do better and meaningful things.  For me, this would be writing my
assignments and completing my degree.

In short, there's no point in learning or understanding the
installation process.  Ideally it should be a simple click in a
graphic installer, but I do not have the time or motivation to
accomplish such integration.

I hope that, by writing the guides, I have simplified things a little
bit, so that people can get things done more quickly, without the
trial-and-error and many midnight debugging sessions that I've had to
endure during the last few years.

**Comment by ray-holmium** [comment]
I agree that a one-click solution would be the dream. My lofty, pie-in-the-sky goal is to one day have zfs support on the official NixOS installer!

If you have time, I would greatly appreciate your criticisms of my deploy script, as it was heavily influenced by your guide. I am particularly interested in starting a dialogue with you about this because I am making some changes to your proposed /etc/nixos layout that may be valuable upstream to your guide as well.

And this is a great learning opportunity! Your guide helped me understand a lot about bash scripting and the structure of NixOS. Thank you for making it available!
